iroh-rpc-client/src/network.rs

```diff
@@ -248,44 +232,61 @@ impl P2pClient {
             topic_hash: topic_hash.to_string(),
             data,
         };
-        let res = self.backend.gossipsub_publish(req).await?;
+        let res = self.client.rpc(req).await??;
         let message_id = MessageId::new(&res.message_id);
         Ok(message_id)
     }
 
     #[tracing::instrument(skip(self))]
     pub async fn gossipsub_remove_explicit_peer(&self, peer_id: PeerId) -> Result<()> {
-        let req = GossipsubPeerIdMsg {
-            peer_id: peer_id.to_bytes(),
-        };
-        self.backend.gossipsub_remove_explicit_peer(req).await?;
+        let req = GossipsubRemoveExplicitPeerRequest { peer_id };
+        self.client.rpc(req).await??;
         Ok(())
     }
 
     #[tracing::instrument(skip(self))]
     pub async fn gossipsub_subscribe(&self, topic: TopicHash) -> Result<bool> {
-        let req = GossipsubTopicHashMsg {
+        let req = GossipsubSubscribeRequest {
             topic_hash: topic.to_string(),
         };
-        let res = self.backend.gossipsub_subscribe(req).await?;
+        let res = self.client.rpc(req).await??;
         Ok(res.was_subscribed)
     }
 
     #[tracing::instrument(skip(self))]
     pub async fn gossipsub_topics(&self) -> Result<Vec<TopicHash>> {
-        let res = self.backend.gossipsub_topics(()).await?;
+        let res = self.client.rpc(GossipsubTopicsRequest).await??;
         let topics = res.topics.into_iter().map(TopicHash::from_raw).collect();
         Ok(topics)
     }
 
     #[tracing::instrument(skip(self))]
     pub async fn gossipsub_unsubscribe(&self, topic: TopicHash) -> Result<bool> {
-        let req = GossipsubTopicHashMsg {
+        let req = GossipsubUnsubscribeRequest {
             topic_hash: topic.to_string(),
         };
-        let res = self.backend.gossipsub_unsubscribe(req).await?;
+        let res = self.client.rpc(req).await??;
         Ok(res.was_subscribed)
     }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn check(&self) -> StatusRow {
+        let status: ServiceStatus = self
+            .version()
+            .await
+            .map(|_| ServiceStatus::Serving)
+            .unwrap_or_else(|_e| ServiceStatus::Unknown);
+        StatusRow {
+            name: "p2p",
+            number: 2,
+            status,
+        }
+    }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn watch(&self) -> impl Stream<Item = StatusRow> {
+        futures::stream::pending()
```

Comment:
sad face

Reply:
My proposal was to deal with this after the merge. See https://github.com/n0-computer/iroh/issues/584